packages/core/src/button/button.element.scss

```diff
@@ -36,6 +36,12 @@
   --color: #{$cds-global-typography-color-100};
 }
 
+:host([action='outline'][status='warning']) ::slotted(cds-badge) {
+  --border-color: #{$cds-alias-status-warning-dark};
+  --background: #{$cds-alias-status-warning-dark};
+  --color: #{$cds-global-typography-color-100};
```

Comment:
The color here I believe would be `--color: #{$cds-global-color-black};` to match the contrast